packages/client/src/components/MediaCard/MediaCardsGrid.tsx

```diff
@@ -1,27 +1,47 @@
-import { DishDto, DishesApi } from '@fullstack/sdk';
-import Grid from '@mui/material/Grid';
+import { DishDto, DishesApi, DishTagsEnum, MealTypeEnum } from '@fullstack/sdk';
+// import { Stack } from '@mui/material';
+import { CircularProgress, Grid, Typography } from '@mui/material';
+import _ from 'lodash';
 import { useEffect, useState } from 'react';
 
 import { apiConfiguration } from '../../config';
+import { SingleFilterType, useFiltersContext } from '../../contexts/FiltersContext';
 import { MediaCard } from './MediaCard';
 
 const dishesApi = new DishesApi(apiConfiguration);
 
 const MediaCardsGrid = () => {
   const [dishes, setDishes] = useState<DishDto[]>([]);
+  const { filters } = useFiltersContext();
+
+  function enumMap(objProperty: (string | null)[], enumName: Record<string, string>) {
+    return objProperty?.map((item) => (item ? enumName[item] : ''));
+  }
 
   useEffect(() => {
-    async function fetchData() {
-      try {
-        const dishReponse = await dishesApi.listAllDishes();
-        setDishes(dishReponse.data);
-      } catch (error) {
-        setDishes([]);
+    let hasChanged = true;
```

Comment:
Czy to `hasChanged` cokolwiek zmienia? 🤔

Jest początkowo ustawione na `true`, po czym dalej jest `if` sprawdzający je (który zawsze się wykona, bo `hasChanged` jest na sztywno ustawiony na `true`). Po czym w cleanupie zmieniamy `hasChanged` na `false`, co nic nie zmienia, bo bezpośrednio po tym callback `useEffect` kończy działanie i przy jego kolejnym wywołaniu `hasChanged` jest znowu ustawiony na sztywno na `true`.

Usunąłem deklarację tej zmiennej, niżej `if`a, który z niej korzysta i cały cleanup i filtrowanie zdaje się działać tak samo jak przed usunięciem tego.

---

Jeżeli celem było zachowanie wartości `hasChanged` między wywołaniami `useEffect` to potrzebny będzie `useState` albo `useRef`, ale nie wiem czy coś takiego jest w ogóle potrzebne bo wydaje mi się, że wszystko działa też bez niego.

Reply:
Hmm napewno miałam sytuacje, że podwójnie wywoływał mi się `useEffect` z nieodpowiednią tablicą z `dishes` i nie wiem już na jakim etapie też miałam _infinite loop_ :D. Rzeczywiście wygląda na to, że nie ma już tego problemu, więc musiałam coś zmienić w międzyczasie. Dzięki :)